server/models/remote_provider.go

```diff
@@ -269,9 +270,16 @@ func (l *RemoteProvider) executePrefSync(tokenString string, sess *Preference) {
 //
 // Every Remote Provider must offer this function
 func (l *RemoteProvider) InitiateLogin(w http.ResponseWriter, r *http.Request, _ bool) {
-	callbackURL := r.Context().Value(MesheryServerCallbackURL).(string)
+	baseCallbackURL := r.Context().Value(MesheryServerCallbackURL).(string)
+
+	refURLqueryParam := r.URL.Query().Get("ref")
```

Comment:
```suggestion
	// Support for deep-link and redirection to land user on their originally requested page post authentication instead of dropping user on the root (home) page.
	refURLqueryParam := r.URL.Query().Get("ref")
```